src/Oscoin/Protocol.hs

```diff
@@ -126,6 +126,14 @@ runProtocol validateFull scoreBlock telemetry bs config use =
         let o = O.emptyOrphanage scoreBlock
         proto               <- newMVar (Protocol bs o validateFull scoreBlock config)
         incomingBlocksQueue <- atomically $ newTBQueue 64
+
+        worker <- Async.async $ forever $ do
+                      block  <- atomically $ readTBQueue incomingBlocksQueue
+                      events <- modifyMVar proto $ \p -> stepProtocol p block
+                      forM_ events (Telemetry.emit telemetry)
+
+        Async.link worker
```

Comment:
Could we document this behavior? “All exceptions raised by the protocol worker are re-thrown to the calling thread“

Reply:
I guess we could, although it's kind-of implicit from the documentation of `link` 😉

Reply:
aac062c9

Reply:
I actually meant documenting this in the function documentation so that you know about this behavior without looking at the code.

Reply:
I see, I see. Ok. 👍

Reply:
e1881017 

As I was at it, I have also partially rewrote the top-level documentation altogether, as it was quite stale.